### PR TITLE
Fix category sorting on PHP8

### PR DESCRIPTION
**Pull request by Nevay**
### 1. Why is this change necessary?
PHP8 changed the behavior of sort functions, [RFC stable sorting](https://wiki.php.net/rfc/stable_sorting) Illegal comparison functions:
> This works, because PHP currently only checks whether the comparison result is “greater than” or not, and never explicitly distinguishes the “equal” and “smaller than” cases. This breaks down with the approach proposed here, because we now do need to know whether values are equal or not, in order to use the fallback sorting criterion.

### 2. What does this change do, exactly?
Return `-1` instead of `0` if left is smaller than right.

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/CategoryGateway.php [hunk 1]
```diff
@@ -120,10 +120,10 @@ public function getList(array $ids, Struct\ShopContextInterface $context)
         //use php usort instead of running mysql order by to prevent file-sort and temporary table statement
         usort($data, function ($a, $b) {
             if ($a['__category_position'] === $b['__category_position']) {
-                return $a['__category_id'] > $b['__category_id'] ? 1 : 0;
+                return $a['__category_id'] > $b['__category_id'] ? 1 : -1;
```
Comment (mitelg) [inline engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/CategoryGateway.php]: we should simply use the space ship operator here
```suggestion
                return $a['__category_id'] <=> $b['__category_id'];
```

File: engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/CategoryGateway.php [hunk 2]
```diff
@@ -120,10 +120,10 @@ public function getList(array $ids, Struct\ShopContextInterface $context)
...
             }
 
-            return $a['__category_position'] > $b['__category_position'] ? 1 : 0;
+            return $a['__category_position'] > $b['__category_position'] ? 1 : -1;
```
Comment (mitelg) [inline engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/CategoryGateway.php]:
```suggestion
            return $a['__category_position'] <=> $b['__category_position'];
```

**Review by mitelg — CHANGES_REQUESTED** [review]

**Comment by mitelg** [comment]
hey @Nevay 

thanks for your first contribution :+1: 

I left some comments, to even improve the code more :slightly_smiling_face:

**Review by mitelg — APPROVED** [review]

**Comment by LePrau** [comment]
The failed mysql:8.0 unit test seems to be unrelated.
Is a re-run needed?

**Comment by mitelg** [comment]
I did a re-run and it works now :slightly_smiling_face:

**Comment by mitelg** [comment]
thanks for your contribution @Nevay  :+1: :tada: :blue_heart:

don't be confused, we merged it on the next branch, which will be merged back to the main branch in a few days

**Pull request closed without merging**